mathesar_ui/src/sections/table-view/Header.svelte

```diff
@@ -88,6 +93,32 @@
       resetPositions: oldSize === 0 || group.size === 0,
     });
   }
+
+  const getColumnNames = () => {
+    const columnNames:string[] = [];
+    const columnData = (columns.data) ? columns.data : null;
+    if (columnData) {
+      columnData.forEach((col:TableColumn) => columnNames.push(col.name));
+    }
+    return columnNames;
+  };
```

Comment:
This method seems to be used only for checking if a column with same name exists. This could probably be changed to something like:

```
  function isColumnPresent(name) {
    return columns.data?.some(col => col.name.toLowerCase() === name?.toLowerCase());
  }
  
  // Computed property to identify duplicate column
  $: isDuplicateColumn = isColumnPresent(newColumnName);
```

Reply:
I have implemented this as suggested.
Thanks!

Reply:
https://github.com/pegnott/mathesar/blob/754414d202a29033bd437f2d845269d5885a5831/mathesar_ui/src/sections/table-view/Header.svelte#L149